### ISSUE TITLE
segfault during hashTableInsert

### ISSUE THREAD

**Issue by chapmanb** [GregoryFaust/samblaster#26]
I'm running into a segfault with samblaster when streaming input from bwa. We've
seen this a few times with different user reports on bcbio but this is the first
reproducible case I've had.

The segfault happens in different positions on two different machines but after
processing to ~7Gb of SAM output it will fail during hashTableInsert with this
traceback from gdb:

```
#0  0x0000000000405212 in hashTableInsert(hashTable*, unsigned long) ()
#1  0x00000000004038d1 in markDupsDiscordants(splitLine*, state_struct*) ()
#2  0x0000000000404107 in processSAMBlock(splitLine*, state_struct*) ()
#3  0x00000000004017b0 in main ()
```

Here is a set of input files and shell script that reproduce the issue for me on
two different machines:
https://s3.amazonaws.com/chapmanb/testcases/samblaster_segfault.tar.gz

Thanks so much for any suggestions or tips to avoid the problem.

**Comment by GregoryFaust** [comment]
Brad, 

Thanks for the reproducible case. I'll take a look at this as soon as possible.

Greg

**Comment by GregoryFaust** [comment]
Fixed in 0.1.24

**Comment by GregoryFaust** [comment]
A more general fix for this issue is in 0.1.25. See release notes for more details.
